maestro-orchestra/src/main/java/maestro/orchestra/Orchestra.kt

```diff
@@ -290,13 +290,19 @@ class Orchestra(
         if (this::jsEngine.isInitialized) {
             jsEngine.close()
         }
-        val shouldUseGraalJs =
-            config?.ext?.get("jsEngine") == "graaljs" || System.getenv("MAESTRO_USE_GRAALJS") == "true"
+        val isRhinoExplicitlyRequested = config?.ext?.get("jsEngine") == "rhino"
+        
+        // Warn users about deprecated Rhino JS engine
+        if (isRhinoExplicitlyRequested) {
+            logger.warn("⚠️  The Rhino JS engine (jsEngine: rhino) is deprecated and will be removed in a future version. Please migrate to GraalJS (the default) for better performance and compatibility.")
+        }
```

Comment:
I don't think this will be shown to users this just redirects to the maestro.log file on debug artifacts.

For showing to users we have Insights abstraction. You can use that to show this warning.